### ISSUE TITLE
PassIFox won´t work with firefox 49.0

### ISSUE THREAD

**Issue by alpine310** [pfn/passifox#555]
Hello
my firefox was updated to version 49.0. Still this day PassiFox won´t work anymore. PassiFox is still show as "installed" in firefox. KeePass version ist still 2.28. KeePassHttp version is still 1.8.3.0.

Are there any know problems with firefox 49.0?
How can I solve this problem?

**Comment by marekbedy** [comment]
I have the same issue, after updating to the firefox version 49.0 it stopped working.
PassiFox v1.2.0
Keepass v2.34
KeePassHttp v1.8.4.1

**Comment by sukhoi47** [comment]
Same problem here, passifox no longer works after update to Firefox 49.

**Comment by agjennings** [comment]
I had to go to Hep -> About Firefox and it updated to 49.0.1 and it is now working again.
Windows 7
Firefox 49.0.1
Passifox 1.2.0
KeePass 2.34
KeepPassHttp 1.8.4.1

**Comment by sukhoi47** [comment]
Here it is working partially on firefox 49.0.1.

Right-click -> Fill User & Pass           _(working)_
Right-click -> Fill Password               _(not working)_

**Comment by purejava** [comment]
Same here for latest Keepass, keepass-plugin-http and Firefox on Arch linux.
I noticed this a couple of days ago as the Arch linux machine is used more frequently.

This morning, I updated my Win 10 machine from Firefox 48.0.2 to 49.0.1 and keepass-plugin-http stopped working.
When I look at the preferences of Firefox and display the stored passwords (which usually displays the contents that are stored in my Keepass database) only a subset of all passswords is displayed. Lots are missing. And the displayed do not work either.

Symptom: when I open an URL with credential fields, keepass-plugin-http does not offer anything. Usually I doubleclick in the Username field, select the appropriate entry and the credentials get filled in.

Now no credentials are offered.

If you need more testing or further info please let me know.

**Comment by purejava** [comment]
I noticed that this issue has been reported [as Issue #528](https://github.com/pfn/passifox/issues/528) already.

**Comment by alireza4050** [comment]
+1

**Comment by dpolivaev** [comment]
+1

**Comment by Minims** [comment]
+1

**Comment by Obusmydealz** [comment]
+1

**Comment by bigfrank16** [comment]
+1

**Comment by alinuxfan** [comment]
As an update, Right click --> Fill Pass is still working but Firefox is still trying to save all passwords even though they are already in KeepPass.

**Comment by mporter-veeva** [comment]
@alinuxfan Not exactly. I have a site that contains multiple credentials. Prior to the firefox update, I could select the username and then the associated password for the unique username would be filled automatically. Now, I am not able to select the username. I must type it and then right click "Fill Pass" and again select the credential password. OR, I can right click on username and select "Fill User & Pass", then scroll through the list and find the credentials I need. Prior to the update, I could select the username (list coming from PassIFox) and then the username and password would automatically fill.

**Comment by alinuxfan** [comment]
@mporter-veeva Ok. My usage, prior to Firefox 49, was to start typing username, get it from the Firefox history, hit TAB, and then let passifox grab the password for me and hit ENTER

**Comment by xiorcal** [comment]
+1

**Comment by dahamsta** [comment]
+1. Getting very tedious. :(

**Comment by androidland** [comment]
+1

**Comment by agsteele** [comment]
My experience exactly matches @alinuxfan

**Comment by pfn** [comment]
If someone puts together a pr to fix this, I will review, merge and release
a new version. At the moment, I have no time to investigate Firefox at all.

**Comment by ibuk01** [comment]
+1

**Comment by dahamsta** [comment]
@pfn, would a bounty help?

**Comment by djgreg13** [comment]
nsLoginManager has changed, only basic auth work, but in Firefox 51, it's terrible, basic auth not working too

**Comment by agsteele** [comment]
@djgreg13 Are you saying that the issue is really a Firefox problem? If so does the issue need to raise with Mozilla?

**Comment by djgreg13** [comment]
I think as a change of Mozilla

I will investigate more if i fnd time :)

Keefox still auto fill

**Comment by djgreg13** [comment]
Found a change in general changelog, we have to diff login manager between two versions
https://www.mozilla.org/en-US/firefox/49.0/releasenotes/

**Comment by WF000** [comment]
Yep, it is somewhat mozilla's 'problem' (assholes, no further notice yaddadaa) , they intercepted login calls to their 'new' login manager ... so therefore making keefox+http inoperable.
If you right click any username box , you'll see new option in context menu ->'Fill login'
..well, that crap broke all autotype. Manual fill user/pass still works.
Otoh, guess, no big deal, rework of plugin needed.. Thanks in advance.

**Comment by djgreg13** [comment]
Keefox is working

Its autofill is different than passifox

Envoyé de mon iPhone

**Comment by byteborg** [comment]
+1

**Comment by WF000** [comment]
@djgreg13 Thanks , that'll do as alternative in meantime, otoh.. click away diff ... imo I find passifox better, optimized, lowweight packed 20kb vs 1mb :D (not quite funny as it looks in overall tdy byte size standards, but clearly seen in programmin standpoint how much code and resources can be avoided to actually damn thing works as exepected ;)

**Comment by Obusmydealz** [comment]
@djgreg13 Thanks, but I tried Keefox and it is no alternative. I could not adjust the default-autofill-entry and I had to click on the Keefox Icon to select which login should be filled in. Passifox did a better integration. I hope someone can fix it.

**Comment by sukhoi47** [comment]
I tried keefox. It is not capable to fill username and password when the login window pops-up and has many other limitations.

**Comment by Mte90** [comment]
The problem exist on firefox 50, is very annoying...

**Comment by pollyzoid** [comment]
The relevant changes on FF49 seem to be these: https://hg.mozilla.org/integration/fx-team/pushloghtml?fromchange=c57a0e71e7cdf62ba3d443e250e6086368ad0833&tochange=f59918cf04fa5954101b0274ebf60fb7e11e7a21 (probably just the ones for bug 667233).

**Comment by gaberad** [comment]
Think I might have found the fix:

Found this commit on a different password integrator:
https://github.com/jfitzell/mozilla-keychain/commit/15682ccf0a44be3732f83e94c9a198260604b2a2

It looks like the background processes are trying use the searchLogins function, but currently searchLogins is just returning an empty array.

I updated the searchLogins function to the following, based on the commit above and it all seems to work correctly (in FF Dev 51.0a2):
```
searchLogins: function searchLogins(count, matchData) {
	this.stub(arguments);
	var host, form, realm;

	try{
		host = matchData.getProperty("hostname");
	} catch(e){
		host = '';
	}
	try{
		form = matchData.getProperty("formSubmitURL");
	} catch(e){
		form = '';
	}
	try{
		realm = matchData.getProperty("httpRealm");
	} catch(e){
		realm = '';
	}

	return this.findLogins(count, host, form, realm);
},
```

Putting in a pull request so you can try this out

**Comment by djgreg13** [comment]
oh yes :) il will try this in my dev edition

2016-10-17 11:46 GMT+02:00 gaberad <notifications@github.com>:

**Comment by djgreg13** [comment]
working, please agree merge et release plugin :)

2016-10-17 11:47 GMT+02:00 Gregory G <greg.djg13@gmail.com>:

**Comment by agsteele** [comment]
Thanks for persisting with this issue.  If I knew had to implement this potential fix I'd try it out. Instead I'll look forward to an updated add-on ;)

**Comment by Anachron** [comment]
Please merge this, my firefox wont allow unsigned addons anymore :(

**Comment by hockeymikey** [comment]
@Anachron Dev version or WaterFox

**Comment by TamasBarta** [comment]
Yesterday the fixes were released. Upgrade your addons. It fixed the issue for me.

**Comment by sukhoi47** [comment]
I just updated and it is working as expected once again.

A big THANKS to the developer @pfn and this active community.

**Comment by alpine310** [comment]
I just updated too and it is working very well with firefox 49
Thanks to the developer

**Comment by agsteele** [comment]
Let me add my thanks for @pfn and @gaberad for resolving this issue.  Much appreciated.

However I'm being prompted to save passwords everytime I log back into a site even if the password is already in KeePass.  Anyone have the same issue or a solution.  Is this a separate issue to post?  It started at the same time that the login problem started.

**Comment by dahamsta** [comment]
Yes, thank you both. Unfortunately, I'm also still being prompted as above.

**Comment by prof-tech** [comment]
Tried to install the updated addon 1.2.1 into released version of FF 49.0.2 and it would not install because it is not signed. Can you make it signed? Really missing the add-on. :-(

**Comment by Obusmydealz** [comment]
Thanks a lot. Now the autofill works, but I've the same issue as @agsteele above.

**Comment by cheuk-fung** [comment]
Thanks a lot. It now works 😄

**Comment by woyczek** [comment]
With MacPass, notifications enabled, FF 49.0.2, PassIFox 1.21.

The notifications says "Request for ie" or "request for com".

It seems the TLD (or the very last part of the URL) is transmitted, but not the whole URL.
To reproduce :
save a password for localhost, or any of your localdomain host ; enable notifications on MacPass or KeePass ; save a password for it, and look at the notifications.
It will work for any hosts without domain, but will fail for any other.

**Comment by pfn** [comment]
Get it from amo

**Comment by pfn** [comment]
Thanks for investigating and resolving this. I have submitted an update to
amo and it should be live once it goes through review and is approved.

**Comment by gaberad** [comment]
@agsteele @dahamsta 
I looked into it and the issue is that when firefox checks if the there is an already existing login for the submitted form, it checks the http realm as part of its checks.

It was failing this check because (based on my testing) the submitted form login httpRealm property is null, but the passifox httpRealm property is an empty string.

The passifox httpRealm is an empty string because of this part of the code in my pull request (based on that other addons commit):
It is entering the catch part, and setting realm to an empty string:
```
	    try{
            realm = matchData.getProperty("httpRealm");
        } catch(e){
            realm = '';
        }
```

I'll put in a pull request where it sets the realm to null if it falls into the catch:

```
	    try{
            realm = matchData.getProperty("httpRealm");
        } catch(e){
            realm = null;
        }
```


I've tried it on a bunch of logins and it worked for all of them for me (it prompts correctly for new/updated logins), but it might be useful if someone tests it out as well.

**Comment by agsteele** [comment]
Thanks @gaberad if you would like to push the compiled xpi in my direction I'd be happy to try it out.  I don't have time to compile a copy for myself just now.

Meanwhile I trust that @pfn will catch the new pr :) in #564 

Appreciate the excellent work by all.

**Comment by hockeymikey** [comment]
@gaberad
Works on my end.

**Comment by j-degreef** [comment]
Hi,
I don't know if something has changed in 1.2.1 causing the "Update password" dialog box to popup each time a login/password is automatically filled and nothing has changed.
I have this problem since a few days, probably since the update to 1.2.1.
Before the automatic filling was not working as stated at the top of this discussion.

**Comment by agsteele** [comment]
@j-degreef A resolution of the issue you describe has been identified and a pull request has been suggested. #564 There is a fuller explanation up this thread.

**Comment by j-degreef** [comment]
@agsteele,  yes, I've just tested #564 and I confirm it solve this problem.
Thanks ;)

**Comment by agsteele** [comment]
@pfn I wonder if you've seen the pr for this issue with PassiFox #564

**Comment by prof-tech** [comment]
Many thanks to pfn, gaberad, and anyone else who looked at this. Installed 1.2.2 on FF 49.0.2 and it looks like everything is working again. PassIFox and KeePass are a must have!

**Comment by j-degreef** [comment]
Great work guys ;)

**Comment by dahamsta** [comment]
Thank you all!
